pkg/gatekeeper/gatekeeper.go

```diff
@@ -53,7 +56,9 @@ func GetGatekeeperEnvironment(httpClient utils.HTTPClient, host string) (*Gateke
 	var environment GatekeeperEnvironment
 	err = json.Unmarshal(byteArray, &environment)
 	if err != nil {
-		return nil, err
+		logr.Trace("Unable to parse response from URL. Bad JSON received: ", err.Error())
+		badJSONErr := errors.New("Bad response received. The URL provided " + host + " should point to the Codewind Gatekeeper service. Check its route " + routePath + " returns valid JSON")
```

Comment:
Something like this may be a bit clearer?
```suggestion
	errMsg := fmt.Sprintf("Bad response received. The URL provided %s should point to the Codewind Gatekeeper service. Check its route %s returns valid JSON", host, routePath)
	badJSONErr := errors.New(errMsg)
```